src/components/PopupDelete.js

```diff
@@ -0,0 +1,25 @@
+import { eventType, selectors } from '../constants/app-data.js';
+import Popup from './Popup.js';
+
+export default class PopupDelete extends Popup {
+	constructor(popupSelector) {
+		super({ popupSelector });
+		this._formElement = this._popupElement.querySelector(
+			selectors.modal.form
+		);
+	}
+
+	close() {
+		this._formElement.removeEventListener(
+			eventType.SUBMIT,
+			this._deleteHandler
+		);
+		super.close();
+	}
+
+	setEventListeners(deleteHandler) {
```

Comment:
* Every popup should be created only one time and `setEventListeners` should be called also once, otherwise you add listeners again and again to the same elements. This can cause a **`memory leak`**


Please, use method `open` for reassigning the function


```js
  open(deleteHandler) {
    this._deleteHandler = deleteHandler;
    super.open();
  }
```